openupgrade_scripts/scripts/sale_timesheet/15.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,120 @@
+from openupgradelib import openupgrade
+
+
+def _pre_value_of_order_id(env):
+    if not openupgrade.column_exists(
+        env.cr,
+        "account_analytic_line",
+        "order_id",
+    ):
+        openupgrade.logged_query(
+            env.cr,
+            """
+            ALTER TABLE account_analytic_line
+            ADD COLUMN order_id integer DEFAULT NULL
+            """,
+        )
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE account_analytic_line
+            SET order_id = sol.order_id
+            FROM
+                account_analytic_line AS aal
+                INNER JOIN sale_order_line AS sol
+                    ON aal.so_line = sol.id
+            WHERE aal.order_id IS NULL
```

Comment:
@daonamutc1 anh xem đoạn này có chạy không?
```suggestion
            UPDATE account_analytic_line aal
            SET order_id = sol.order_id
            FROM sale_order_line sol
            WHERE aal.so_line IS NOT NULL AND aal.so_line = sol.id
```